src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -260,27 +265,46 @@ extension RiskProvider: RiskProviding {
 		let group = DispatchGroup()
 
 		group.enter()
-		var appConfiguration: SAP_ApplicationConfiguration?
+
 		appConfigurationProvider.appConfiguration { [weak self] result in
 			switch result {
 			case .success(let config):
-				appConfiguration = config
-
-				self?.determineSummary(userInitiated: userInitiated, ignoreCachedSummary: ignoreCachedSummary, appConfiguration: config) {
-					summary = $0
-					group.leave()
-				}
+				self?.determineSummary(
+					userInitiated: userInitiated,
+					ignoreCachedSummary: ignoreCachedSummary,
+					appConfiguration: config,
+					completion: { result in
+						guard let self = self else { return }
+
+						switch result {
+						case .success(let summary):
+							self._requestRiskLevel(
```

Comment:
I'm not sure if this has any practical relevance, but by moving this up here this call also falls under the 8 minute limit. I guess there is no time-heavy work done from this point on, but I'm not sure and just want to make sure you were aware of that implication.

Reply:
Good point! Moved it out again.